snuba/redis.py

```diff
@@ -31,24 +39,44 @@ def execute_command(self, *args: Any, **kwargs: Any) -> Any:
             return super(self.__class__, self).execute_command(*args, **kwargs)
 
 
-redis_client: RedisClientType
-
-if settings.USE_REDIS_CLUSTER:
-    startup_nodes = settings.REDIS_CLUSTER_STARTUP_NODES
-    if startup_nodes is None:
-        startup_nodes = [{"host": settings.REDIS_HOST, "port": settings.REDIS_PORT}]
-    redis_client = RetryingStrictRedisCluster(
-        startup_nodes=startup_nodes,
-        socket_keepalive=True,
-        password=settings.REDIS_PASSWORD,
-        # HACK(mattrobenolt): See https://github.com/Grokzen/redis-py-cluster/pull/353
-        max_connections_per_node=True,
-    )
-else:
-    redis_client = StrictRedis(
-        host=settings.REDIS_HOST,
-        port=settings.REDIS_PORT,
-        password=settings.REDIS_PASSWORD,
-        db=settings.REDIS_DB,
-        socket_keepalive=True,
-    )
+def _retry(func: Callable[[], RedisClientType]) -> Callable[[], RedisClientType]:
+    @wraps(func)
+    def wrapper() -> RedisClientType:
+        retry_counter = 3
+        while retry_counter > 0:
+            try:
+                return func()
+            except RedisClusterException as e:
+                if "All slots are not" in str(e) and retry_counter >= 0:
```

Comment:
I don't love text matching in an error message but the stakes are low here if we can't check a class type or against a constant